third_party/dml/DirectMLX.h

```diff
@@ -2644,6 +2670,48 @@ namespace dml
         return output;
     }
 
+    inline Expression BatchNormalizationGrad(
+        Expression inputX,
+        Expression inputYBackprop,
+        Expression mean,
+        Expression variance,
+        Expression scale,
+        float epsilon,
+        Expression &outputScaleGradient,
```

Comment:
For operators that return multiple tensors, you should return a struct rather than using (ref!) out params. See GRU or RandomGenerator for example.